### PR TITLE
core: api to save/get app settings to migrate them as part of the database

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: docs/rfcs/2024-02-19-settings.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+- put settings to the database via the API
+- save settings as some file with cross-platform format (e.g. JSON or YAML or properties used on desktop).
+
+The second approach seems much simpler than maintaining the settings in the database.
+
+If we save a file, then there are two options:
+- native apps maintain cross-platform schemas for this file, support any JSON and parse it in a safe way (so that even invalid or incorrect JSON - e.g., array instead of object - or invalid types in some properties do not cause the failure of properties that are correct).
+- this schema and type will be maintained in the core library, that will be responsible for storing and reading the settings and passing to native UI as correct record of a given type.
+
+The downside of the second approach is that addition of any property that needs to be migrated will have to be done on any change in either of the platforms. The downside of the first approach is that neither app platform will be self-sufficient any more, and not only iOS/Android would have to take into account code, but also each other code.
```
Comment (spaced4ndy) [inline docs/rfcs/2024-02-19-settings.md:28]: even with second approach iOS/Android still have to take into account each others' code, no? In a sense that settings different per OS will be making it into AppSettings type, for example different call settings, appearance settings.

How will settings that couldn't be applied (but were successfully parsed) be treated? E.g. migrating Android to iOS back to Android will preserve Android specific settings or they will be forgotten?

Comment (epoberezkin) [reply docs/rfcs/2024-02-19-settings.md:28]: yes, will be dropped

File: docs/rfcs/2024-02-19-settings.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+If we go with the second approach, there will be these types:
+
+```haskell
+data AppSettings = AppSettings
+  { version :: Version, -- settings file schema version, so we can migrate settings to another schema if necessary
+    networkConfig :: NetworkConfig, -- existing type in Haskell and all UIs
+    privacyConfig :: PrivacyConfig -- new type, etc.
```
[the remark this answers is not in the document]
Comment (epoberezkin) [reply docs/rfcs/2024-02-19-settings.md:35]: no

File: src/Simplex/Chat/AppSettings.hs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+      privacyShowChatPreviews = Just True,
+      privacySaveLastDraft = Just True,
+      privacyProtectScreen = Just False,
+      webrtcPolicyRelay = Just True,
+      webrtcICEServers = Just [],
+      callKitCallsInRecents = Nothing,
+      confirmRemoteSessions = Just False,
+      connectRemoteViaMulticast = Just True,
+      connectRemoteViaMulticastAuto = Just True,
+      developerTools = Just False
```
Comment (spaced4ndy) [inline src/Simplex/Chat/AppSettings.hs:56]: + confirm database upgrades?

File: src/Simplex/Chat/AppSettings.hs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      networkConfig = Just defaultNetworkConfig,
+      privacyEncryptLocalFiles = Just True,
+      privacyAcceptImages = Just True,
+      privacyLinkPreviews = Just True,
+      privacyShowChatPreviews = Just True,
+      privacySaveLastDraft = Just True,
+      privacyProtectScreen = Just False,
+      webrtcPolicyRelay = Just True,
+      webrtcICEServers = Just [],
+      callKitCallsInRecents = Nothing,
```
Comment (spaced4ndy) [inline src/Simplex/Chat/AppSettings.hs:52]: + use iOS call interface?

File: src/Simplex/Chat/AppSettings.hs [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    developerTools :: Maybe Bool
+  }
+  deriving (Show)
+
+defaultAppSettings :: AppSettings
+defaultAppSettings =
+  AppSettings
+    { appPlatform = Nothing,
+      networkConfig = Just defaultNetworkConfig,
+      privacyEncryptLocalFiles = Just True,
```
Comment (spaced4ndy) [inline src/Simplex/Chat/AppSettings.hs:44]: \+ notifications preview?

File: src/Simplex/Chat/AppSettings.hs (hunk 4 above)
Comment (spaced4ndy) [reply src/Simplex/Chat/AppSettings.hs:52]: + calls on lock screen for android?

**Pull request merged**